### ISSUE TITLE
Add new field to skeletal elements to capture count

### ISSUE THREAD

**Issue by TarawaProject** [spawaskar-cora/cora-docs#182]
Add new field to skeletal elements to capture count. For example unidentified fragments 15, distal manual phalanges 131.
<img width="395" alt="elements count" src="https://user-images.githubusercontent.com/33848426/41125018-57e2af64-6a3f-11e8-8033-76a060209121.PNG">

**Comment by SachinPawaskarUNO** [comment]
@TarawaProject, Please verify, this has currently been implemented for the following bones

1. Unidentified cranium
2. Unidentified fragments
3. Unidentified rib fragments
4. Unidentified long bone
5. Unidentified metacarpal/tarsal
6. Unidentified phalanx
7. Unidentified vertebra

Adding @TarawaProject, @CTegtmeyer, @clegarde @skindsc1 and @emilystreetman to this conversation. Please note that the count and mass fields become available only after you Save/Create the record. See workflow below.

![image](https://user-images.githubusercontent.com/19599773/48914843-418fde80-ee42-11e8-91ca-97fb943e15e8.png)

![image](https://user-images.githubusercontent.com/19599773/48915045-deeb1280-ee42-11e8-86e4-1cf8dfdfa996.png)

**Comment by CTegtmeyer** [comment]
I have verified that this has been implemented for these elements. Before I close this issue, can I open the discussion (@fedamann, @SachinPawaskarUNO) of adding counts to smaller elements like phalanges so that we do not have to enter each individual phalanx to CoRA? Would this be something to consider, or should they keep being individually put in (I only ask because we sometimes get upwards of 100+ phalanges in an accession and each one has to be individually entered right now).

**Comment by SachinPawaskarUNO** [comment]
@CTegtmeyer and @fedamann This issue is related to #296 where I have mentioned the following. Currently only the list of Skeletal_Bones (SB) flagged as countable will have count and mass fields available for inventorying purposes. We currently have Unidentified phalanx that could be use for this purpose. We could consider renaming this to **Unidentified phalanges**.
Anyway, I would like the groups in Hawaii & Omaha to work together on coming up with a list of such bones, if what we have currently is not sufficient. We currently have the following bones that have these capabilities

Unidentified cranium
Unidentified fragments
Unidentified rib fragments
Unidentified long bone
Unidentified metacarpal/tarsal
Unidentified phalanx
Unidentified vertebra

**Comment by TarawaProject** [comment]
What the Tarawa project is looking for is a count for "distal manual phalanges" not just unidentified phalanx.  If we can agree on a business rule that they are treated as unidentified phalanx I am ok with this, but think this would cause more confusion than necessary.  I think this needs to be a discussion topic @CTegtmeyer @Megyesimm  for the scrum

**Comment by SachinPawaskarUNO** [comment]
@fedamann, @TarawaProject, @CTegtmeyer We do not have "distal manual phalanges" as a bone in CoRA, we have distal manual phalanx 1,..., distal manual phalanx 5 and distal pedal phalanx 1,..., distal pedal phalanx 5. But these in my opinion should be used for a specific instance of a bone and not a collection of bones.

As I think through this a couple of thoughts come to my mind.
1. We cannot just have "distal manual phalanges" because in that case we should also have "distal pedal phalanges", I would reckon that having just "distal phalanges" would suffice.
2. Why just distal? What about intermediate and proximal phalanges?

The reason I was suggesting using "Unidentified Phalanx" by renaming it to "Unidentified Phalanges" is that then we already have it and don't have to create new bones is CoRA. It also accounts for manual/pedal as well as distal/intermediate/proximal, unless it would be useful to have both "Unidentified Phalanx" and "Unidentified Phalanges" as bones in CoRA.

Anyway please give this some deliberation in the coming days and lets discuss this on Friday (Nov 30th 2018) when I'm at DPAA and make a decision, because I would really like the Tarawa project to have this capability.

**Comment by clegarde** [comment]
I think the reason to have distal manual phalanges, intermediate manual phalanges, etc. would be if the project wants to separate these counts. We do have "unseriated distal/intermediate/proximal manual/pedal phalanx", so I think we could add the count option to these to take care of this.

SIDE COMMENT/THOUGHT: would it be possible to add a function to the "new skeletal element" page that is a "count" but it would generate that many new skeletal elements of that same bone? Like how the bone grouping can give designators to each element in the bone group? For example, your new skeletal element is "unseriated distal manual phalanx", starting designator (like in bone group) is 1000 and then if you say the count is 100, you'd get 100 new elements with designators 1000, 1001, 1002, etc.? This would be in cases that the project would want every element in the inventory, but it could allow projects to enter all of one bone at a time (particularly for phalanges), and in certain cases, this could be more efficient. Particularly for non-DPAA cases or cases without much provenience. They might want to do all 30 left humeri, then 40 left radii, etc. Just a thought...please ignore if already discussed or not applicable now.

**Comment by SachinPawaskarUNO** [comment]
@fedamann @TarawaProject @CTegtmeyer @clegarde @emilystreetman This is good conversation, please continue this with the end goal of making a recommendation by this Friday for what we want to do for the CoRA v1.0.0 release.

**Comment by CTegtmeyer** [comment]
I agree that I think @TarawaProject was referring more towards the categories of unseriated distal, intermediate, and proximal phalanges (both manual and pedal -- at least this is how I understood it in my discussion with Rebecca).

I think it would be okay to have these smaller bones be countable and not have analytics attached to them. Unseriated phalanges don't really offer much in the way of analytics, and we rarely sample them for DNA, but I like @clegarde suggestion to have the ability to add a batch import as well, especially if we could retain analytical power over the imported/batched bones.

**Comment by SachinPawaskarUNO** [comment]
@fedamann @TarawaProject @CTegtmeyer @clegarde @emilystreetman If we agree to have these smaller bones be countable and not have analytics then can we agree on the list below.

1. unseriated proximal manual phalanges
2. unseriated intermediate manual phalanges
3. unseriated distal manual phalanges
4. unseriated proximal pedal phalanges
5. unseriated intermediate pedal phalanges and
6. unseriated distal pedal phalanges

Please note that batch import is separate for what we are discussing here. Please do not mix these issues. We are planning to eventually add a batch import capability to CoRA starting with Skeletal Elements which can have analytics tied to it.

**Comment by clegarde** [comment]
Yes, I will clarify that I was not talking about a batch import. I was just suggesting another way to generate multiple new elements at one time into CoRA. Similar to adding a "bone group" but for it to be generating multiple entries/designators of the same skeletal element. Just thinking about what may be more efficient entry for different projects/scenarios.

**Comment by SachinPawaskarUNO** [comment]
So, @clegarde do you agree with the list above in context of the current need for the Tarawa project?

**Comment by clegarde** [comment]
Yes, I agree that the above list should meet the needs for the Tarawa project.

**Comment by SachinPawaskarUNO** [comment]
@emilystreetman Thank you.
@CTegtmeyer @TarawaProject @fedamann Let me know you thoughts ASAP. 
For now based on the conversation above and my understanding I'm recommending we flag the following bones as countable for use with the Tarawa project. 

1. unseriated distal manual phalanx
2. unseriated intermediate manual phalanx
3. unseriated proximal manual phalanx
4. unseriated distal pedal phalanx
5. unseriated intermediate pedal phalanx
6. unseriated proximal pedal phalanx

In the future, we may also want to consider the following as countable. Thoughts?

1. unseriated cervical vertebra
2. unseriated lumbar vertebra
3. unseriated metacarpal
4. unseriated metatarsal
5. unseriated rib
6. unseriated thoracic vertebra

**Comment by emilystreetman** [comment]
I am verifying the phalanges as countable and making a new ticket for the suggested list.
